webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js

```diff
@@ -9,42 +12,47 @@ import {
   ModalFooter,
 } from '../../../../../commonComponents/modal'
 
-export default class DeleteSurveyDialog extends React.Component {
+class DeleteSurveyDialog extends React.Component {
   constructor (props) {
     super(props)
 
-    this.state = {confirmName: ''}
+    this.state = { confirmName: '' }
     this.confirmNameOnChange = this.confirmNameOnChange.bind(this)
     this.confirmNameRef = React.createRef()
   }
 
   componentDidMount () {
-    this.setState({confirmName: ''})
+    this.setState({ confirmName: '' })
     this.confirmNameRef.current.focus()
   }
 
   confirmNameOnChange (event) {
-    this.setState({confirmName: event.target.value})
+    this.setState({ confirmName: event.target.value })
   }
 
   render () {
-    const {surveyName, onDelete, onCancel} = this.props
+    const { surveyName, onDelete, onCancel } = this.props
+    const { i18n } = this.context
```

Comment:
Can we use useContext here as well?

Reply:
Apparently hooks don't work inside classes. We can try switch to a functional component and use the useState hook

Reply:
yes.